internal/app/singularity/pull.go

```diff
@@ -311,6 +351,17 @@ func OciPull(imgCache *cache.Handle, name, imageURI, tmpDir string, ociAuth *oci
 	}
 	if !exists {
 		sylog.Infof("Converting OCI blobs to SIF format")
+
+		// monitor for OS signals and remove invalid file
+		c := make(chan os.Signal)
+		signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+		go func(fileName string) {
```

Comment:
Maybe we should start considering a general signal handler implementation. This is very similar to the signal handling in `build.go` and we can probably create a package to do what we want. I wouldnt let this block the PR, but make a tech debt issue if you dont want to address it in this PR